src/Admin/AbstractAdmin.php

```diff
@@ -1297,6 +1297,12 @@ public function getTemplate($name)
     public function getNewInstance()
     {
         $object = $this->getModelManager()->getModelInstance($this->getClass());
+        if (!is_a($object, $this->getClass(), true)) {
+            throw new \TypeError(
+                sprintf('An instance of %s is expected, %s given', $this->getClass(), \get_class($object))
```

Comment:
Since in your first commit, `null` was a possible value, I think we must update the message to something like this:
```suggestion
                sprintf('An instance of `%s` is expected, `%s` given.', $this->getClass(), \is_object($object) ?  \get_class($object) : \gettype($object))
```

Reply:
@phansys I hesitated. 
I initially handled the `null` case since our tests was returning `null`.
But when I looked at the phpDoc from getModelInstance, it should return an object, `null` is not a valid value. So I didn't know if I should handle this case. With these informations, do you think I still should ?

Reply:
Regardless the phpDoc, I'd prefer an accurate exception message, like the one used when a method argument doesn't respect the required type (https://3v4l.org/v9Gan):
>Argument 1 passed to method() must be an object, null given

Reply:
It's updated @phansys